app/dao/status.go

```diff
@@ -0,0 +1,58 @@
+package dao
+
+import (
+	"context"
+	"fmt"
+	"yatter-backend-go/app/domain/object"
+	"yatter-backend-go/app/domain/repository"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type (
+	// Implementation for repository.Status
+	status struct {
+		// sqlx.DB: database/sql.DBをラップし、より簡単に使えるようにしたもの
+		db *sqlx.DB
+	}
+)
+
+var _ repository.Status = (*status)(nil)
+
+// NewStatus : Create status repository
+// var _ repository.Status = (*status)(nil) で、status が repository.Status インターフェースを実装しているかをチェックしている
+// なので、返り値の型を *status ではなく repository.Status になる
+func NewStatus(db *sqlx.DB) repository.Status {
+	return &status{db: db}
+}
```

Comment:
```suggestion
// NewStatus : Create status repository
// var _ repository.Status = (*status)(nil) で、status が repository.Status インターフェースを実装しているかをチェックしている
// なので、返り値の型を *status ではなく repository.Status になる
func NewStatus(db *sqlx.DB) *status {
	return &status{db: db}
}
```

理由がコメントに書かれていてすごく良いです！
補足として、呼び出し側で型を合わせられるよう、元の型を返す方が良いこともあります 🙋‍♂️ 
（今回は別のインターフェースに入れる予定が無いので問題ないです）


`*status` で返すと呼び出し側で型を選べます
```go
var status1 repository.Status = dao.NewStatus(db)
var status2 any = dao.NewStatus(db)

// *statusがio.Writerを実装していた場合
var status3 io.Writer = dao.NewStatus(db)
```

逆に、repository.Statusで返した場合、io.Writerとして扱えないです